--- miniaudio.h.orig	2025-01-14 10:15:10.205276000 +0300
+++ miniaudio.h	2024-03-04 13:21:13.378473400 +0300
@@ -59155,7 +59155,7 @@
         result = ma_SetFilePointerEx((HANDLE)file, liDistanceToMove, NULL, dwMoveMethod);
     } else if (ma_SetFilePointer != NULL) {
         /* No SetFilePointerEx() so restrict to 31 bits. */
-        if (origin > 0x7FFFFFFF) {
+        if (liDistanceToMove.QuadPart > 0x7FFFFFFF) {
             return MA_OUT_OF_RANGE;
         }
 
